vortex-array/src/flatten.rs

```diff
@@ -17,10 +35,189 @@ pub enum Flattened {
     Primitive(PrimitiveArray),
     Struct(StructArray),
     VarBin(VarBinArray),
-    VarBinView(VarBinViewArray),
+    // TODO(aduffy): VarBinView is being disabled until execution engines improve their
+    //  support for them, or we build better execution kernels of our own.
+    //  Should re-enable once DataFusion completes support for them, tracked in
+    //  https://github.com/apache/datafusion/issues/10918
+    // VarBinView(VarBinViewArray),
     Extension(ExtensionArray),
 }
 
+impl Flattened {
+    /// Convert a flat array into its equivalent [ArrayRef](Arrow array).
+    ///
+    /// Scalar arrays such as Bool and Primitive flattened arrays though should convert with
+    /// zero copies, while more complex variants such as Struct may require allocations if its child
+    /// arrays require decompression.
+    pub fn into_arrow(self) -> ArrayRef {
+        match self {
+            Flattened::Null(a) => null_to_arrow(a),
+            Flattened::Bool(a) => bool_to_arrow(a),
+            Flattened::Primitive(a) => primitive_to_arrow(a),
+            Flattened::Struct(a) => struct_to_arrow(a),
+            Flattened::VarBin(a) => varbin_to_arrow(a),
+            Flattened::Extension(a) => match a.id().as_ref() {
+                "vortex.localdatetime" => local_date_time_to_arrow(
+                    LocalDateTimeArray::try_from(&a.into_array()).expect("localdatetime"),
+                ),
+                _ => panic!("unsupported extension dtype with ID {}", a.id().as_ref()),
```

Comment:
For now. We should later be able to convert into Arrow extension array.

Reply:
ack